libraries/AP_RangeFinder/RangeFinder.cpp

```diff
@@ -705,6 +705,18 @@ void RangeFinder::Log_RFND()
     AP::logger().WriteBlock(&pkt, sizeof(pkt));
 }
 
+bool RangeFinder::prearm_healthy(char *failure_msg, const uint8_t failure_msg_len) const
+{
+    for (uint8_t i = 0; i < RANGEFINDER_MAX_INSTANCES; i++) {
+        if ((params[i].type != RangeFinder_TYPE_NONE) && (drivers[i] == nullptr)) {
+          hal.util->snprintf(failure_msg, failure_msg_len, "Rangefinder %d was not detected", i + 1);
+          return false;
+        }
+    }
```

Comment:
you could add a second check that the status is ok ! see : https://github.com/ArduPilot/ardupilot/pull/9886/files#diff-03adf2804cc10b90bbfe953b5c07b33eR91 
it would catch correct instanciation but late deconnexion !

Reply:
NoData is a valid state though if you are out of range, but don't know which direction you are out of range.